activerecord/lib/active_record/core.rb

```diff
@@ -609,6 +613,24 @@ def strict_loading?
       @strict_loading
     end
 
+    SIBLINGSIZELIMIT = 5000
+
+    def _create_root_load_tree_node(siblings:)
+      return unless ActiveRecord.load_tree_enabled?
+      if siblings.length > ActiveRecord::Core::SIBLINGSIZELIMIT
+        return _create_load_tree_node(creator: self, siblings: [], sibling_size_limit_exceeded: true).set_records
+      end
+      _create_load_tree_node(creator: self, siblings: siblings).set_records
+    end
+
+    def _create_association_load_tree_node(parent:, siblings:, child_name:)
+      return unless ActiveRecord.load_tree_enabled?
+      if siblings.length > SIBLINGSIZELIMIT
```

Comment:
> On the other hand, I don't think we can have a structure that grows unbounded (this is already a problem we sometimes run into and have worked around with the query cache). But IMO if we're only selectively enabling the load tree (for a block?) that may make it acceptable?

@jhawthorn I thought the same thing. I put in a hard upper limit for how big the siblings for a single node can be. I think there is a point where a developer is working with soo much data, they probably need to start thinking about how to use batches or n+1 becomes a better choice perhaps?

 I am not sure about this number, but if the sibling tree is empty, then the tool will fall back to not doing dynamic includes.